src/query/storages/fuse/src/fuse_table.rs

```diff
@@ -525,17 +525,27 @@ impl Table for FuseTable {
         self.do_analyze(&ctx).await
     }
 
-    fn table_statistics(&self) -> Result<Option<TableStatistics>> {
-        let s = &self.table_info.meta.statistics;
-        Ok(Some(TableStatistics {
-            num_rows: Some(s.number_of_rows),
-            data_size: Some(s.data_bytes),
-            data_size_compressed: Some(s.compressed_data_bytes),
-            index_size: Some(s.index_data_bytes),
-        }))
+    async fn table_statistics(&self) -> Result<Option<TableStatistics>> {
+        let res = if let Some(snapshot) = self.read_table_snapshot().await? {
```

Comment:
why not set the `statistics` into `table_info` during create from snapshot?

This changes could lead `read_table_snapshot` to be called every time.